optuna_dashboard/ts/components/GraphParetoFront.tsx

```diff
@@ -165,6 +165,71 @@ const makeMarker = (
   }
 }
 
+const getIsDominatedND = (normalizedValues: number[][]) => {
+  // Fallback for straight-forward pareto front algorithm (O(N^2) complexity).
+  const isDominated: boolean[] = []
+  normalizedValues.forEach((values0: number[]) => {
+    const dominated = normalizedValues.some((values1: number[]) => {
+      if (values0.every((value0: number, k: number) => values1[k] === value0)) {
+        return false
+      }
+      return values0.every((value0: number, k: number) => values1[k] <= value0)
+    })
+    isDominated.push(dominated)
+  })
+  return isDominated
+}
+
+const getIsDominated2D = (normalizedValues: number[][]) => {
+  // Fast pareto front algorithm (O(N log N) complexity).
+  const sorted = normalizedValues
+    .map((values, i) => [values[0], values[1], i])
+    .sort((a, b) =>
+      a[0] > b[0]
+        ? 1
+        : a[0] < b[0]
+        ? -1
+        : a[1] > b[1]
+        ? 1
+        : a[1] < b[1]
+        ? -1
+        : 0
+    )
+  let maxValue0 = sorted[0][0]
+  let minValue1 = sorted[0][1]
```

Comment:
This change is optional, if you think it is better, please take it, but otherwise, please just ignore it!
```suggestion
  let maxValueSeen0 = sorted[0][0]
  let minValueSeen1 = sorted[0][1]
```